rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp

```diff
@@ -460,15 +456,8 @@ TEST_F(TestSubscriptionTopicStatisticsFixture, test_receive_stats_for_message_no
   EXPECT_EQ(kNumExpectedMessagePeriodMessages, message_period_count);
 
   // Check the collected statistics for message period.
-  // Message age statistics will not be calculated because Empty messages
-  // don't have a `header` with timestamp. This means that we expect to receive a `message_age`
-  // and `message_period` message for each empty message published.
   for (const auto & msg : received_messages) {
-    if (msg.metrics_source == kMessageAgeSourceLabel) {
-      check_if_statistics_message_is_empty(msg);
-    } else if (msg.metrics_source == kMessagePeriodSourceLabel) {
-      check_if_statistic_message_is_populated(msg);
-    }
+    check_if_statistic_message_is_populated(msg);
```

Comment:
If we're removing much of these statistics due to presumably not needing the `header` part of messages, then how many tests should we have for messages. For instance we have both 
[`test_receive_stats_for_message_with_header`](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp#L475-L520)
and
[`test_receive_stats_for_message_no_header`](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp#L410-L473)

I would assume that we won't need cases for header and without header since it doesn't do anything related to `topic_statistics` now. Now we can leave a lot of the logic and just remove the builky tests, but we can also use `rclcpp/msg/string.hpp` opposed to `rclcpp/msg/message_with_header.hpp` since the string will have data outside of the timestamp

Reply:
@CursedRock17 This is a good question that I overlooked.
As regards to the  [test_receive_stats_for_message_with_header](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp#L475-L520) I think we can remove it completely since the timestamps in `message_info` will be settled up automatically on the DDS layer and we can't control them and the `test_receive_stats_for_message_with_header` will become effectively the same as [test_receive_stats_for_message_no_header](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp#L410-L473).
However, I still don't know what to do with the [test_receive_stats_include_window_reset](https://github.com/ros2/rclcpp/blob/7f7ffcf6ed1143b4d4a9e0cf248e190731b2e5f8/rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp#L532-L588). 
We can't fix it or replace it effectively since we can't control the time stamp on the publisher side anymore.

Reply:
It seems I've got an idea of how to rewrite [test_receive_stats_include_window_reset](https://github.com/ros2/rclcpp/blob/7f7ffcf6ed1143b4d4a9e0cf248e190731b2e5f8/rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp#L532-L588). I will try..

Reply:
@CursedRock17 I've workarounded [test_receive_stats_include_window_reset](https://github.com/ros2/rclcpp/blob/7f7ffcf6ed1143b4d4a9e0cf248e190731b2e5f8/rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp#L532-L588) by using publisher emulator instead of real publisher. I am calling the `rclcpp::Subscription::handle_message(msg_shared_ptr, message_info)` directly with modified `message_info` instead of publishing messages via a real publisher. :stuck_out_tongue_winking_eye: 
Also made a major cleanup in the subscription topic statistics tests in a separate commit.
The PR is ready for the next round of review.

Reply:
Yup, removing all of the extra specifiers seems to work and they didn't seem necessary anyways. As for the `message_emulator` as long as we can test the `handle_message` function in the context of a subscription and it passes these tests, I don't see any problem why we couldn't do it. 

These changes look good to me.

Reply:
@CursedRock17 If you don't have any other concerns could you please approve this PR?

Reply:
@MichaelOrlov it should be approved now